README.md

```diff
@@ -1,81 +1,104 @@
-- Unify configuration of various cloud compute providers
-- Automatically destroy unused cloud resources (compute instances are terminated on job completion/failure, and storage is removed when results are downloaded)
+- **Provision Resources**: create cloud compute (CPU, GPU, RAM) & storage resources without reading pages of documentation
+- **Sync & Execute**: easily sync & run local data & code in the cloud
+- **Low cost**: transparent auto-recovery from interrupted low-cost spot/preemptible instances
+- **No waste**: auto-cleanup unused resources (terminate compute instances upon job completion/failure & remove storage upon download of results)
+- **No lock-in**: switch between several cloud vendors with ease due to concise unified configuration
 
-The Iterative Provider can provision resources with the following cloud providers and orchestrators:
+Supported cloud vendors include:
 
-- Amazon Web Services
+- Amazon Web Services (AWS)
 - Microsoft Azure
-- Google Cloud Platform
-- Kubernetes
+- Google Cloud Platform (GCP)
+- Kubernetes (K8s)
 
-## Documentation
+## Usage
 
-See the [Getting Started](https://registry.terraform.io/providers/iterative/iterative/latest/docs/guides/getting-started) guide to learn how to use the Iterative Provider. More details on configuring and using the Iterative Provider are in the [documentation](https://registry.terraform.io/providers/iterative/iterative/latest/docs).
+### Requirements
 
-## Support
+- [Install Terraform 1.0+](https://learn.hashicorp.com/tutorials/terraform/install-cli#install-terraform), e.g.:
+  + Brew (Homebrew/Mac OS): `brew tap hashicorp/tap && brew install hashicorp/tap/terraform`
+  + Choco (Chocolatey/Windows): `choco install terraform`
+  + Conda (Anaconda): `conda install -c conda-forge terraform`
+  + Debian (Ubuntu/Linux):
+    ```
+    sudo apt-get update && sudo apt-get install -y gnupg software-properties-common curl
+    curl -fsSL https://apt.releases.hashicorp.com/gpg | sudo apt-key add -
+    sudo apt-add-repository "deb [arch=amd64] https://apt.releases.hashicorp.com $(lsb_release -cs) main"
+    sudo apt-get update && sudo apt-get install terraform
+    ```
+- Create an account with any supported cloud vendor and expose its [authentication credentials via environment variables](https://registry.terraform.io/providers/iterative/iterative/latest/docs/guides/authentication)
 
-Have a feature request or found a bug? Let us know via [GitHub issues](https://github.com/iterative/terraform-provider-iterative/issues). Have questions? Join our [community on Discord](https://discord.gg/bzA6uY7); we'll be happy to help you get started!
+### Define a Task
 
-## License
+In a project root directory, create a file named `main.tf` with the following contents:
 
-Iterative Provider is released under the [Apache 2.0 License](https://github.com/iterative/terraform-provider-iterative/blob/master/LICENSE).
-
-## Development
-
-### Install Go 1.17+
```

Comment:
I was thinking... what if a user does not have any ml training script on heirs hands? Or a user has a script but the environment settings will prevent user from executing it? Does it make sense to provide a script in additional to the TF file? 

It should reduce the entrance bar quite significantly.

An ideal tutorial should have all the commands, scripts and data I need to run to get a result. Example - dvc tutorial.

Reply:
> user does not have any ml training script

I don't follow, do you mean #305?

> user has a script but the environment settings will prevent executing

Do you mean the script won't execute locally or won't execute on the cloud? And why won't it execute? Because of missing env vars such as `NUM_EPOCHS` or something? Or missing dependencies like `numpy`?

Reply:
I mean, user might not have any ml training script at all (or it won't work).
In DVC tutorial, user downloads code as the first step. Do we need something similar here?

Reply:
ah so an example repo? No we don't have one (yet). Technically we'd probably need 3 - one each for AWS/Azure/GCP.

Reply:
> _we'd probably need 3 - one each for AWS/Azure/GCP_

😬 Unless we use [iterative/example-repos-dev](https://github.com/iterative/example-repos-dev) or similar, it doesn't look like the best idea from a maintainability standpoint.

Reply:
Yes, the only required change is `cloud = "whatever"` 😅

Reply:
I would think at a minimum:

- `main.tf`: `cloud = "X"`
- `README.md`: "How to set `export X_CREDENTIALS`"

Plus probably:

- `requirements.txt`
- `run.py`
- `.github/workflows/cml.yaml`

Reply:
don't think this complexity is needed in the readme here. Right now it's a self-contained[^1] example that supports users both with and without a script file.

[^1]: apart from the how-to-setup-credentials external link